components/compositing/constellation.rs

```diff
@@ -665,16 +669,18 @@ impl<LTF: LayoutThreadFactory, STF: ScriptThreadFactory> Constellation<LTF, STF>
             }
             Request::Script(FromScriptMsg::ForwardMouseButtonEvent(
                     pipeline_id, event_type, button, point)) => {
-                if let Some(pipeline) = self.pipelines.get(&pipeline_id) {
-                    pipeline.script_chan.send(ConstellationControlMsg::SendEvent(pipeline_id,
-                        CompositorEvent::MouseButtonEvent(event_type, button, point))).unwrap();
-                }
+                let event = CompositorEvent::MouseButtonEvent(event_type, button, point);
+                let msg = ConstellationControlMsg::SendEvent(pipeline_id, event);
+                self.pipelines.get(&pipeline_id)
+                    .and_then(|pipeline| pipeline.script_chan.send(msg).ok())
+                    .unwrap_or_else(|| self.handle_send_error(pipeline_id));
```

Comment:
FWIW I find this very wrong and I wouldn't r+. We have a tradition of not using and_then and unwrap_or_else for side-effectful code.